### PR TITLE
Refactor SpawnRequest to be composed of RouterConfig and ExecutorConfig

### PR DESCRIPTION
**Pull request by pretentious7**
https://github.com/drifting-in-space/plane/issues/161

I also tried a bunch of things with trying an Executor trait with a trait object, but that felt bad,. However I think this is a good start

### CODE REVIEW

File: core/src/messages/agent.rs [hunk 1]
```diff
@@ -254,6 +255,49 @@ pub struct SpawnRequest {
...
+    pub credentials: Option<DockerCredentials>,
+
+    /// Resource limits
+    #[serde(default = "ResourceLimits::default")]
+    pub resource_limits: ResourceLimits,
+}
+
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct RouterConfig {
```
Comment (paulgb) [inline core/src/messages/agent.rs:283]: I'm inclined to call this `ProxyConfig`, and use it for things that are specific to the proxy. Currently, that's just the `backend_id`, but will include auth tokens (#118) eventually as well.

We could also just leave these fields in the `SpawnRequest` itself -- unlike the executor, I don't see a clear advantage to bundling them together at this point.

**Review by paulgb — COMMENTED** [review]
Thanks for this, left a couple comments.

File: core/src/messages/agent.rs [hunk 2]
```diff
@@ -254,6 +255,49 @@ pub struct SpawnRequest {
...
+
+#[serde_as]
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+pub struct DockerExecutableConfig {
+    /// The container image to run.
+    pub image: String,
+
+    /// The timeout after which the drone is shut down if no connections are made.
+    #[serde_as(as = "DurationSeconds")]
+    pub max_idle_secs: Duration,
```
[the remark this answers is not in the document]
Comment (pretentious7) [reply core/src/messages/agent.rs:268]: ah that's right, yes will do

File: core/src/messages/agent.rs (hunk 1 above)
Comment (pretentious7) [reply core/src/messages/agent.rs:283]: I think that's sensible yes, will do (leaving the other fields in the SpawnRequest itself). The current main motivation after all is to have `DockerExecutableConfig` be seperately serializeable.

**Review by paulgb — APPROVED** [review]
LGTM, thanks for the changes.

**Pull request merged**